### PR TITLE
Feature/vaccinecer 2073 refactoring controller

### PR DESCRIPTION
**Pull request by haraldloesing**

### CODE REVIEW

File: src/main/java/ch/admin/bag/covidcertificate/service/CovidCertificatePdfGenerationService.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+import static ch.admin.bag.covidcertificate.api.Constants.CREATE_BARCODE_FAILED;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class CovidCertificatePdfGenerationService {
```
Comment (Armin-Isenring-Bit) [inline src/main/java/ch/admin/bag/covidcertificate/service/CovidCertificatePdfGenerationService.java:32]: To me a class name containing fromExisting would be more accurate. By reading the class name I'd know what endpoints are handled.

File: src/main/java/ch/admin/bag/covidcertificate/web/controller/CovidCertificatePdfGenerateController.java [hunk 2]
```diff
@@ -25,7 +25,7 @@
 @Slf4j
 public class CovidCertificatePdfGenerateController {
```
Comment (Armin-Isenring-Bit) [inline src/main/java/ch/admin/bag/covidcertificate/web/controller/CovidCertificatePdfGenerateController.java:26]: Same as with the service: to me a class name containing fromExisting would be more accurate.

File: src/main/java/ch/admin/bag/covidcertificate/web/controller/CovidCertificateGenerationController.java [hunk 3]
```diff
@@ -158,6 +160,15 @@ public CovidCertificateCreateResponseDto createMedicalExemptionCertificate(
         log.info("Call of create for exceptional certificate");
 
         createDto.validate();
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            objectMapper.findAndRegisterModules();
+            objectMapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+            String demo = objectMapper.writeValueAsString(createDto);
+            log.info("Moinsen demo is: {}", demo);
```
[the remark this answers is not in the document]
Comment (haraldloesing) [reply src/main/java/ch/admin/bag/covidcertificate/web/controller/CovidCertificateGenerationController.java:168]: Sorry that was during debugging.

File: src/test/java/ch/admin/bag/covidcertificate/web/controller/CovidCertificatePdfGenerationControllerSecurityTest.java [hunk 4]
```diff
@@ -63,13 +63,13 @@ void setupMocks() throws IOException {
         //                any(RecoveryCertificateCreateDto.class))).thenReturn(
```
Comment (Armin-Isenring-Bit) [inline src/test/java/ch/admin/bag/covidcertificate/web/controller/CovidCertificatePdfGenerationControllerSecurityTest.java:63]: remove unnecessary code.

Comment (haraldloesing) [reply src/test/java/ch/admin/bag/covidcertificate/web/controller/CovidCertificatePdfGenerationControllerSecurityTest.java]: done

**Review by Armin-Isenring-Bit — APPROVED** [review]

**Pull request merged**